base/applications/notepad/dialog.c

```diff
@@ -283,24 +283,20 @@ int GetSelectionText(HWND hWnd, LPTSTR lpString, int nMaxCount)
 static RECT
 GetPrintingRect(HDC hdc, RECT margins)
 {
-    int iLogPixelsX, iLogPixelsY;
-    int iHorzRes, iVertRes;
-    int iPhysPageX, iPhysPageY, iPhysPageW, iPhysPageH;
+    INT iLogPixelsX = GetDeviceCaps(hdc, LOGPIXELSX);
+    INT iLogPixelsY = GetDeviceCaps(hdc, LOGPIXELSY);
+    INT iHorzRes = GetDeviceCaps(hdc, HORZRES); /* in pixels */
+    INT iVertRes = GetDeviceCaps(hdc, VERTRES); /* in pixels */
     RECT rcPrintRect;
 
-    iPhysPageX = GetDeviceCaps(hdc, PHYSICALOFFSETX);
-    iPhysPageY = GetDeviceCaps(hdc, PHYSICALOFFSETY);
-    iPhysPageW = GetDeviceCaps(hdc, PHYSICALWIDTH);
-    iPhysPageH = GetDeviceCaps(hdc, PHYSICALHEIGHT);
-    iLogPixelsX = GetDeviceCaps(hdc, LOGPIXELSX);
-    iLogPixelsY = GetDeviceCaps(hdc, LOGPIXELSY);
-    iHorzRes = GetDeviceCaps(hdc, HORZRES);
-    iVertRes = GetDeviceCaps(hdc, VERTRES);
-
-    rcPrintRect.left = (margins.left * iLogPixelsX / 2540) - iPhysPageX;
-    rcPrintRect.top = (margins.top * iLogPixelsY / 2540) - iPhysPageY;
-    rcPrintRect.right = iHorzRes - (((margins.left * iLogPixelsX / 2540) - iPhysPageX) + ((margins.right * iLogPixelsX / 2540) - (iPhysPageW - iPhysPageX - iHorzRes)));
-    rcPrintRect.bottom = iVertRes - (((margins.top * iLogPixelsY / 2540) - iPhysPageY) + ((margins.bottom * iLogPixelsY / 2540) - (iPhysPageH - iPhysPageY - iVertRes)));
+#define CONVERT_X(x) MulDiv((x), iLogPixelsX, 2540) /* 100th millimeters to pixels */
```

Comment:
Any reason why you left out `PHYSICALOFFSETX` ?

Reply:
I thought these are needless in actual printing.
But the physical margins should seem be checked before printing. The logical margins of the current settings should be larger than the physical margins.

Reply:
Adjusted.